### PR TITLE
fix dead links in shims.md [skip ci]

### PR DESCRIPTION
**Pull request by nvliyuan**
fix dead links in shims.md

### CODE REVIEW

File: docs/dev/shims.md [hunk 1]
```diff
@@ -43,8 +43,8 @@ between Spark 3.1.x and Spark 3.2.x. So instead of deriving from such classes di
 inject an intermediate trait e.g. `com.nvidia.spark.rapids.shims.ShimExpression` that
 has a varying source code depending on the Spark version we compile against to overcome this
 issue as you can see e.g., comparing TreeNode:
-1. [ShimExpression For 3.0.x and 3.1.x](https://github.com/NVIDIA/spark-rapids/blob/main/sql-plugin/src/main/post320-treenode/scala/com/nvidia/spark/rapids/shims/v2/TreeNode.scala#L23)
-2. [ShimExpression For 3.2.x](https://github.com/NVIDIA/spark-rapids/blob/main/sql-plugin/src/main/pre320-treenode/scala/com/nvidia/spark/rapids/shims/v2/TreeNode.scala#L23)
+1. [ShimExpression For 3.0.x and 3.1.x](https://github.com/NVIDIA/spark-rapids/blob/main/sql-plugin/src/main/post320-treenode/scala/com/nvidia/spark/rapids/shims/TreeNode.scala#L23)
```
Comment (firestarman) [inline docs/dev/shims.md:46]:
```suggestion
1. [ShimExpression For 3.0.x and 3.1.x](https://github.com/NVIDIA/spark-rapids/blob/main/sql-plugin/src/main/pre320-treenode/scala/com/nvidia/spark/rapids/shims/TreeNode.scala#L23)
```

**Comment by pxLi** [comment]
build

File: docs/dev/shims.md [hunk 2]
```diff
@@ -43,8 +43,8 @@ between Spark 3.1.x and Spark 3.2.x. So instead of deriving from such classes di
...
+2. [ShimExpression For 3.2.x](https://github.com/NVIDIA/spark-rapids/blob/main/sql-plugin/src/main/pre320-treenode/scala/com/nvidia/spark/rapids/shims/TreeNode.scala#L23)
```
Comment (firestarman) [inline docs/dev/shims.md:47]:
```suggestion
2. [ShimExpression For 3.2.x](https://github.com/NVIDIA/spark-rapids/blob/main/sql-plugin/src/main/post320-treenode/scala/com/nvidia/spark/rapids/shims/TreeNode.scala#L23)
```

**Review by firestarman — COMMENTED** [review]
LGTM

**Review by GaryShen2008 — APPROVED** [review]

**Comment by nvliyuan** [comment]
build

**Pull request merged**